tools/migrate_chart/migrate_chart.py

```
@@ -0,0 +1,93 @@
+HARBOR_CA_PATH = Path('/harbor_ca.crt')
+
+class ChartV2:
+
+    def __init__(self, filepath:Path):
+        self.filepath = filepath
+        self.project = self.filepath.parts[-2]
+        parts = self.filepath.stem.split('-')
+        flag = False
+        for i in range(len(parts)-1, -1, -1):
+            if parts[i][0].isnumeric():
+                self.name, self.version = '-'.join(parts[:i]), '-'.join(parts[i:])
+                flag = True
+                break
+        if not flag:
+            raise Exception('chart name is illegal')
+
+    def __check_exist(self, hostname, username, password):
+        return requests.get(CHART_URL_PATTERN.format(
+                host=hostname,
+                project=self.project,
+                name=self.name,
+                version=self.version),
+                auth=requests.auth.HTTPBasicAuth(username, password))
+
+    def migrate(self, hostname, username, password):
+        res = self.__check_exist(hostname, username, password)
+        if res.status_code == 200:
+            raise Exception("Artifact already exist in harbor")
+        if res.status_code == 401:
+            raise Exception(res.reason)
+
+        oci_ref = "{host}/{project}/{name}:{version}".format(
+            host=hostname,
+            project=self.project,
+            name=self.name,
+            version=self.version)
+
+        return subprocess.run([MIGRATE_CHART_SCRIPT, HELM_CMD, self.filepath, oci_ref],
+        text=True, stdout=subprocess.DEVNULL, stderr=subprocess.PIPE)
+
+
+@click.command()
+@click.option('--hostname', default='127.0.0.1', help='the password to login harbor')
+@click.option('--username', default='admin', help='The username to login harbor')
+@click.option('--password', default='Harbor12345', help='the password to login harbor')
+def migrate(hostname, username, password):
+    """
+    Migrate chart v2 to harbor oci registry
+    """
+    subprocess.run([CA_UPDATE_CMD])
```

Fail early if the user is not an admin?